tests/test-grammar-integration.cpp

```diff
@@ -28,6 +28,18 @@ static llama_grammar* build_grammar(const std::string & grammar_str) {
     return grammar;
 }
 
+static bool test_build_grammar_fails(const std::string & grammar_str) {
+    bool grammar_fails = false;
+    try {
+        build_grammar(grammar_str);
+        fprintf(stderr, "❌ Expected build failure, but succeeded: %s\n", grammar_str.c_str());
+    } catch (const std::exception & err) {
+        grammar_fails = true;
+        fprintf(stdout, "✅︎\n");
+    }
+    return grammar_fails;
+}
+
```

Comment:
Minor formatting adjustment so that we print the grammar on success as well.
```suggestion
static bool test_build_grammar_fails(const std::string & grammar_str) {
    fprintf(stderr, "⚫ Testing failure for grammar: %s\n", grammar_str.c_str());

    bool grammar_fails = false;
    try {
        build_grammar(grammar_str);
        fprintf(stderr, "  ❌ Expected build failure, but succeeded\n");
    } catch (const std::exception & err) {
        grammar_fails = true;
        fprintf(stdout, "  ✅︎\n");
    }
    return grammar_fails;
}
```